src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -71,4 +85,95 @@ public enum JsonTokenType : byte
         public bool TryGetInt64Value(out long value) { throw null; }
         public bool TryGetSingleValue(out float value) { throw null; }
     }
+    public ref partial struct Utf8JsonWriter
+    {
+        private object _dummy;
+        public Utf8JsonWriter(System.Buffers.IBufferWriter<byte> bufferWriter, System.Text.Json.JsonWriterState state = default(System.Text.Json.JsonWriterState)) { throw null; }
+        public long BytesCommitted { get { throw null; } }
+        public long BytesWritten { get { throw null; } }
+        public int CurrentDepth { get { throw null; } }
+        public void Flush(bool isFinalBlock = true) { }
+        public System.Text.Json.JsonWriterState GetCurrentState() { throw null; }
+        public void WriteBoolean(System.ReadOnlySpan<byte> propertyName, bool value, bool escape = true) { }
+        public void WriteBoolean(System.ReadOnlySpan<char> propertyName, bool value, bool escape = true) { }
+        public void WriteBoolean(string propertyName, bool value, bool escape = true) { }
+        public void WriteBooleanValue(bool value) { }
+        public void WriteCommentValue(System.ReadOnlySpan<byte> value, bool escape = true) { }
+        public void WriteCommentValue(System.ReadOnlySpan<char> value, bool escape = true) { }
+        public void WriteCommentValue(string value, bool escape = true) { }
+        public void WriteEndArray() { }
+        public void WriteEndObject() { }
+        public void WriteNull(System.ReadOnlySpan<byte> propertyName, bool escape = true) { }
+        public void WriteNull(System.ReadOnlySpan<char> propertyName, bool escape = true) { }
+        public void WriteNull(string propertyName, bool escape = true) { }
+        public void WriteNullValue() { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, decimal value, bool escape = true) { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, double value, bool escape = true) { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, int value, bool escape = true) { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, long value, bool escape = true) { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, float value, bool escape = true) { }
```

Comment:
I still feel that the most consistent story (across the new Json types and the rest of the framework) here would be that "string" and "ReadOnlySpan&lt;char&gt;" don't need an encoding in their name, but that "ReadOnlySpan&lt;byte&gt;" does.

For Utf8Json**Reader** the "what are these input bytes?" didn't need to say the encoding, because it was in the class name.  Likewise, for the Reader, the "give me the raw bytes" APIs didn't need to say the encoding, because it was in the class name... they're both talking about the same input source.

For the Writer, the Utf8 in the name means that when it sends bytes to the IBufferOutput that it will send UTF-8 data.  That does not mean that this property is assuming UTF-8 (because now the bytes are input, but the UTF-8 in the type name is about output).  This property could easily be written to do content-sniffing (not that I like content sniffing); but putting "utf8" in the name then makes it very clear that this is for UTF-8 data.

Essentially, these overloads take "ReadOnlySpan&lt;byte&gt;" only because Utf8String doesn't exist, therefore they want to say "utf8PropertyName".  The ones on the Reader, on the other hand, would still have been ReadOnlySpan&lt;byte&gt; because they are about "the wire representation".

Reply:
I'm not sure what change you're asking for here. Could you make this more concrete. I agree your general stance, tho.

Reply:
After talking to @bartonjs it seems he's proposing that we name the parameter `utf8PropertyName` because this is an input, not an output, so the class name doesn't really apply. I'm OK with that.